### PR TITLE
fix: Added Links for all Charts in Preview & Published Mode

### PR DESCRIPTION
**Pull request by noahonyejese**
This PR fixes Issue #1813 

### How to Test
1. Go to this [Link](https://visualization-tool-azw7yjcr8-ixt1.vercel.app) 
2. Publish, a tab layout see how "Created with..." is visible ✅
3. Publish a Dashboard Layout with 1 Chart see how "Created with..." is in the Chart ✅
4. Publish a Dashboard Layout with more than 1 Chart see how the "Created with..." is only at the bottom of the layout ✅

### CODE REVIEW

File: app/components/chart-preview.tsx [hunk 1]
```diff
@@ -568,6 +568,7 @@ const ChartPreviewInner = ({
                   chartConfig={chartConfig}
                   dashboardFilters={state.dashboardFilters}
                   components={allComponents}
+                  showVisualizeLink
```
Comment (bprusinowski) [inline app/components/chart-preview.tsx:571]: @noahonyejese I think we should only show this when a chart is published, the current behavior here is correct; see how it looks in the design, e.g. here: https://www.figma.com/design/Hn9wvuEYLUmdJtg4KwZYUx/visualize.admin.ch---design?node-id=2805-554265&m=dev

File: app/components/chart-published.tsx [hunk 2]
```diff
@@ -443,8 +447,9 @@ const ChartPublishedInnerImpl = (props: ChartPublishInnerProps) => {
               dashboardFilters={state.dashboardFilters}
               components={allComponents}
               showVisualizeLink={
-                state.chartConfigs.length === 1 &&
-                state.layout.type !== "dashboard"
+                state.layout.type !== "dashboard" ||
+                (state.layout.type === "dashboard" &&
+                  state.chartConfigs.length === 1)
```
Comment (bprusinowski) [inline app/components/chart-published.tsx:452]: 👍

**Review by bprusinowski — COMMENTED** [review]
Thanks @noahonyejese, I think we shouldn't show the `Created with Visualize` link during editing as is the case now on PROD 👀 

Also, the` How to test` link doesn't work (I think you copied a `/create` link that points to a configurator state stored in your `localStorage`) – here one trick is to copy a link as shown in the screenshot that opens the editor mode for a given dataset 👍 

Screenshot 2024-11-06 at 12 48 24

**Comment by noahonyejese** [comment]
@bprusinowski So not show at all for any chart during preview / edit mode ?

**Comment by bprusinowski** [comment]
Yes, the issue was only that the link was missing in case of published dashboards 👍 So in editing mode we never show the link, and in published mode we show it inside individual chart if it's not a dashboard or a dashboard with one chart, or outside of dashboard group for dashboards with more than one chart 😄

**Comment by bprusinowski** [comment]
I see that @sosiology mentioned in the issue that it's not shown in the editing mode, but I am pretty sure we have never shown it here, and that could lead to some issues – if you click it, you'd go out of the editing mode and go back to the home page? 🤔 And we also don't have it in the design, so I'd scope this only to published charts (or we can discuss together 👍)

**Review by bprusinowski — APPROVED** [review]
LGTM, thanks! 💯 

Btw, I noticed that you used a deployment link based on a commit in `How to test` instructions (https://visualization-tool-azw7yjcr8-ixt1.vercel.app/) – to make sure it stays up to date, you should use a deployment link based on a PR (https://visualization-tool-git-fix-created-reference-dashboards-ixt1.vercel.app/en) that updates automatically (I tried to test again after the most recent commit, but the deployment was not updated) :) This might be important for other people that test after some time 👍 

Screenshot 2024-11-06 at 13 11 14

**Comment by noahonyejese** [comment]
> LGTM, thanks! 💯
> 
> Btw, I noticed that you used a deployment link based on a commit in `How to test` instructions (https://visualization-tool-azw7yjcr8-ixt1.vercel.app/) – to make sure it stays up to date, you should use a deployment link based on a PR (https://visualization-tool-git-fix-created-reference-dashboards-ixt1.vercel.app/en) that updates automatically (I tried to test again after the most recent commit, but the deployment was not updated) :) This might be important for other people that test after some time 👍
> 
> Screenshot 2024-11-06 at 13 11 14

oh damn, thanks a lot!

**Pull request merged**